File: ch3/cs/UVa11565.py
def main():
    N = int(input())
    for _ in range(N):
        A,B,C = map(int, input().split())
        sol = False
        for x in range(-22, 23):
            if sol:
                break
            if x*x <= C:
                for y in range(-100, 101):
                    if sol:
                        break
                    if y != x and x*x + y*y <= C:
                        for z in range(-100, 101):
                            if sol:
                                break
                            if z != x and z != y and x+y+z == A and x*y*z == B and x*x + y*y + z*z == C:
                                print("{} {} {}".format(x, y, z))
                                sol = True
        if not sol:
            print("No solution.")

main()

File: ch3/cs/UVa11565.java
import java.util.*;

class Main { /* UVa 11565 - Simple Equations */
  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);
    int N = sc.nextInt();
    while (N-- > 0) {
      int A = sc.nextInt(), B = sc.nextInt(), C = sc.nextInt();
      Boolean sol = false;
      int x, y, z;
      for (x = -22; (x <= 22) && !sol; ++x) if (x*x <= C)
        for (y = -100; (y <= 100) && !sol; ++y) if ((y != x) && (x*x + y*y <= C))
          for (z = -100; (z <= 100) && !sol; ++z)
            if ((z != x) && (z != y) &&
                (x+y+z == A) && (x*y*z == B) && (x*x + y*y + z*z == C)) {
              System.out.printf("%d %d %d\n", x, y, z);
              sol = true;
            }
      if (!sol) System.out.println("No solution.");
    }
  }
}
